src/classes/GS_AdminSetup.cls

```diff
@@ -0,0 +1,375 @@
+        List<GS_Admin_Checklist_Item__mdt> items = [SELECT Id 
+                                                    FROM   GS_Admin_Checklist_Item__mdt
+                                                    WHERE  Id = :itemId];
+        return !items.isEmpty();
+    }
+
+    /**
+     * @description Set an Checklist element as checked to the current user.
+     * It's only done if the element was previously marked as "incomplete"(record does not exists). 
+     * It will not create duplicated records.
+     * @param  adminStatus List<GetStartedCompletionChecklistState__c> list with the adminStatus filtered by user and filter id
+     * @param  itemId      String the item id to be completed
+     */
+    private static void markItemAsCompleted(List<GetStartedCompletionChecklistState__c> adminStatus, String itemId) {
+        if (Schema.SObjectType.GetStartedCompletionChecklistState__c.fields.Item_Id__c.isUpdateable() &&
+            Schema.SObjectType.GetStartedCompletionChecklistState__c.fields.User__c.isUpdateable() && 
+            adminStatus.isEmpty()) {
+            
+            adminStatus.add(new GetStartedCompletionChecklistState__c(Item_Id__c = itemId,
+                                                                     User__c = UserInfo.getUserId()));
+            insert adminStatus;
+        }
+    }
+
+    /**
+     * @description unmark the previously completed checklist item marking it as "incomplete"
+     * In this implementation, we only store the "completed" items so we deleted.
+     * @param  adminStatus List<GetStartedCompletionChecklistState__c> list with the items filtered by currentUser and itemId
+     */
+    private static void unmarkItemAsCompleted(List<GetStartedCompletionChecklistState__c> adminStatus) {
+        if (GetStartedCompletionChecklistState__c.SObjectType.getDescribe().isDeletable() && !adminStatus.isEmpty()) {
+            database.delete(adminStatus);
+        }
+    }
+
+    /**
+     * @description Return the list of GS_Admin_Checklist_Section__mdt and his Items 
+     * @return List<GS_Admin_Checklist_Section__mdt>
+     * @note [ORDER BY Position__c ASC] doesn't work well in the inner query, I need to sort outside SOQL
+     */
+    private static List<GS_Admin_Checklist_Section__mdt> getSectionsAndSubsections() {
+        return [ 
+            SELECT Id, Title_Label__c, Description_Label__c, Position__c, 
+                (
+                    SELECT Id, Title_Label__c, Description_Label__c, Image__c, Position__c,
+                        Has_Link__c, Link_Label__c, Link_URL__c, Extra_Info_Label__c,
+                        Primary_Button_Type__c, Primary_Button_Label__c, Primary_Button_Value__c,
+                        Secondary_Button_Type__c, Secondary_Button_Label__c, Secondary_Button_Value__c
+                    FROM GS_Admin_Checklist_Items__r
+                ) 
+            FROM GS_Admin_Checklist_Section__mdt ORDER BY Position__c ASC 
```

Comment:
Since it was stated that this page is meant for admins only, let's include the `WITH_SECURITY_ENFORCED` clause on this query. It'll be worth testing how the page renders for users without the appropriate permissions.

Reply:
I dont think it is the perfect timing to add changes in Apex code taking in consider we are in the last day with US team support this week, Yisel did the entire regression today. Doing this would mean running cautious QA process and I am afraid of lacking of time. I would do it as a separate WI without blocking this merge. I dont think doing could create a security issue taking in consider the page will only be rendered to Admins and the information that the query returns is less than meaningful